### ISSUE TITLE
Workflow Forms Not Shown When Launched from Bio Block Actions in Certain Circumstances

### ISSUE THREAD

**Issue by billdeitrick** [SparkDevNetwork/Rock#5861]
### Description

In 16.4, workflow forms are not being shown when launched from the Person profile Bio Block's actions menu in certain cases. Instead, the message "This Work does not currently require your attention" is shown.

This only seems to happen with a specific pattern of workflow actions occurring prior to the form. This is an obscure bug, but has broken workflows that were known to be working in v13 and v14 and were not modified since time they were known to be working and an upgrade to 16.4 was completed.

There are a few workarounds, including immediately persisting the workflow and redirecting to the workflow entry page with appropriate parameters.

### Actual Behavior

Launch an affected workflow from the Bio block.
![Snag_7925ac](https://github.com/SparkDevNetwork/Rock/assets/4174527/280ed760-0f64-4707-9137-1191d361661d)

The "...does not require your attention" message is shown.
![Snag_796266](https://github.com/SparkDevNetwork/Rock/assets/4174527/e8f66a92-a177-488a-88f0-a87a470e5cd2)

### Expected Behavior

The workflow form should be presented to the user immediately after launching the workflow from the actions menu.

### Steps to Reproduce

1. Import this workflow

[Workflow Form Test_202405070712.json](https://github.com/SparkDevNetwork/Rock/files/15236871/Workflow.Form.Test_202405070712.json)

2. Add the workflow to the actions menu on the person bio block
3. Launch the workflow from the person bio block
4. Observe the "...does not require your attention" message is shown.

Triggering this bug seems to require the conditionals in the pattern shown in the example workflow. It also seems to be related to use of the Attribute from Entity action. This workflow is a minimal example to reproduce the bug pulled from a much larger workflow.

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16.4

### Client Culture Setting

en-US

**Comment by chead4** [comment]
A new property called "Complete Action If Criteria Unmet" has been added to WorkflowActionType to fix this issue. The default setting is unchecked.  When "Complete Action If Criteria Unmet" is checked,  the Workflow Action will be marked as completed even if the workflow criteria are not met. The workflow log will note that the workflow action was not processed but marked as completed. 

When "Complete Action If Criteria Unmet" is not checked, Workflows Actions with conditional actions remain active forever if one or more conditional action criteria are never met. This would leave the action open to re-examination and execution at a later time, out of sequence.

![chrome_4zDtPPvoHN](https://github.com/user-attachments/assets/ab1d0073-4163-457e-972e-b2548957a6ff)

**Comment by RufenachtPW** [comment]
I wanted to add a little more information to this. I ran into a similar situation where the person was passed in through a page parameter (Launched from the business profile, Page Parameter is "Person"). The attribute from entity action is initially skipped because we have already have the person. The specify situation was checking to see if the "person" was a business or person. If it was a business we moved that to a different attribute and cleared the person field. The problem happened when we try to display the form. The workflow went back and reran the action because the criteria no longer was not met. Which then caused the message that the work does not require your attention. 
![](https://files.slack.com/files-pri/T1BLX1RD2-F07GY64N2F2/screenshot_2024-08-09_at_9.03.46___am.png)
